### PR TITLE
Adds examples from the Design Pattern book for the Decorator Pattern.

### PR DESCRIPTION
**Pull request by priyapatil1**
@c-j-j @dirv

### CODE REVIEW

File: src/main/DarkRoast.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package main;
+
+public class DarkRoast extends Beverage {
+    public DarkRoast() {
+        description = "Most Excellent Dark Roast ";
```
Comment (c-j-j) [inline src/main/DarkRoast.java:5]: I'd make this `this.description =` to make it clear it's setting the field.

File: src/main/StarbuzzApp.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package main;
+
+public class StarbuzzApp {
+    public static void main (String args[]) {
```
Comment (c-j-j) [inline src/main/StarbuzzApp.java:4]: I personally would rather see unit tests to verify the behaviour of the design patterns, as opposed to setting up a main file like this. It'll be easier for us to see what the output of `getDescription()`

Comment (priyapatil1) [inline src/main/StarbuzzApp.java:4]: 👍

File: src/main/DarkRoast.java (hunk 1 above)
Comment (priyapatil1) [inline src/main/DarkRoast.java:5]: 👍

**Pull request merged**